src/BlazorWebView/src/Maui/RootComponentsCollection.cs

```diff
@@ -12,6 +13,7 @@ public class RootComponentsCollection : ObservableCollection<RootComponent>, IJS
 		/// Initializes a new instance of <see cref="RootComponentsCollection"/>.
 		/// </summary>
 		/// <param name="jsComponents">Configuration to enable JS component support.</param>
+		[DynamicDependency(DynamicallyAccessedMemberTypes.NonPublicMethods, typeof(JSComponentConfigurationStore))]
```

Comment:
This seems like a bug in ASP.NET. Consumers (like Maui is here) shouldn't need to add `DynamicDependency` attributes into internal things in ASP.NET in order for trimming to work correctly.

Can we get this fixed in ASP.NET in 7.0?

Looking at https://github.com/dotnet/aspnetcore/blob/d5f487900782d7d66fb106578577467ef9666e21/src/Components/Web/src/WebRenderer.cs#L118-L119, the comment says:

> // Linker preserves this if you call RootComponents.Add

Is no one calling RootComponents.Add?

Reply:
This is waht I'm looking into right onw.

Reply:
Can you point me to where the reflection against `WebRendererInteropMethods.AddRootComponent` occurs? That should help identify who should be responsible for ensuring the method is there.

Reply:
Ok, this PR can probably be closed. The sample was using [this method](https://github.com/dotnet/aspnetcore/blob/dac643de45478622b4cf10e955b605d5e2630925/src/Components/Web/src/JSComponents/JSComponentConfigurationExtensions.cs#L45) for registering the custom element, but it should have been using [this overload](https://github.com/dotnet/aspnetcore/blob/dac643de45478622b4cf10e955b605d5e2630925/src/Components/Web/src/JSComponents/JSComponentConfigurationExtensions.cs#L56). I've confirmed that using the correct overload fixes the problem.

Reply:
See: https://github.com/dotnet/maui/blob/dc4f18da0796ccd45ae60f1d92eeb97d2bcc592d/src/Controls/samples/Controls.Sample/Pages/Blazor/BlazorPage.xaml.cs#L17

Reply:
Does this require any changes in the template project, or do we just need to update the sample in that case with the appropriate overload?

Reply:
Template projects are fine, this is just a case of the `Maui.Controls.Sample` project using the wrong `RegisterForJavaScript` overload. That said, I'm having a discussion offline with @eerhardt about potential improvements to the framework in this area.

Reply:
> Can we get this fixed in ASP.NET in 7.0?

I've opened https://github.com/dotnet/aspnetcore/pull/41610 to fix this in ASP.NET in 7.0.